client/router.js

```diff
@@ -36,12 +37,20 @@ Router.prototype.route = function(path, options) {
   };
 
   this._routes.push(route);
+  if (options.name) {
+    this._pathsMap[options.name] = path;
```

Comment:
We need to when handling middlewares. When using `except` and `only` kind of hooks. So, I prefer to layout the foundation. No need to change name related code later on.